### PR TITLE
Adding clear logging including debug info (GPU)

### PR DESCRIPTION
**Pull request by juhlee-microsoft**
1. No functional change
2. Added more debug msg

Test results of CUDA driver
```
[Canonical UbuntuServer 18.04-LTS latest] [LISAv2 Test Results Summary]
[Canonical UbuntuServer 18.04-LTS latest] Test Run On           : 11/12/2019 23:10:24
[Canonical UbuntuServer 18.04-LTS latest] ARM Image Under Test  : Canonical : UbuntuServer : 18.04-LTS : latest
[Canonical UbuntuServer 18.04-LTS latest] Initial Kernel Version: 5.0.0-1023-azure
[Canonical UbuntuServer 18.04-LTS latest] Final Kernel Version  : 5.0.0-1023-azure
[Canonical UbuntuServer 18.04-LTS latest] Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
[Canonical UbuntuServer 18.04-LTS latest] Total Time (dd:hh:mm) : 0:0:22
[Canonical UbuntuServer 18.04-LTS latest] 
[Canonical UbuntuServer 18.04-LTS latest]    ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
[Canonical UbuntuServer 18.04-LTS latest] -------------------------------------------------------------------------------------------------------------------------------------------
[Canonical UbuntuServer 18.04-LTS latest]     1 GPU                  NVIDIA-CUDA-DRIVER-VALIDATION                                                     PASS                19.83 
[Canonical UbuntuServer 18.04-LTS latest] 	Using nVidia driver : CUDA 
[Canonical UbuntuServer 18.04-LTS latest] 	lsvmbus: Expected "PCI Express pass-through" count: 4, count inside the VM: 4 : PASS 
[Canonical UbuntuServer 18.04-LTS latest] 	lspci: Expected "3D controller: NVIDIA Corporation" count: 4, found inside the VM: 4 : PASS 
[Canonical UbuntuServer 18.04-LTS latest] 	lshw: Expected Display adapters: 4, total adapters found in VM: 4 : PASS 
[Canonical UbuntuServer 18.04-LTS latest] 	nvidia-smi: Expected GPU count: 4, found inside the VM: 4 : PASS 
[Canonical UbuntuServer 18.04-LTS latest] 
[Canonical UbuntuServer 18.04-LTS latest] 
[Canonical UbuntuServer 18.04-LTS latest] Logs can be found at C:\LISAv2\GZ90\TestResults\2019-12-11-23-10-20-9773
```
Test result of GRID driver
```
[Canonical UbuntuServer 18.04-LTS latest] [LISAv2 Test Results Summary]
[Canonical UbuntuServer 18.04-LTS latest] Test Run On           : 11/12/2019 23:10:02
[Canonical UbuntuServer 18.04-LTS latest] ARM Image Under Test  : Canonical : UbuntuServer : 18.04-LTS : latest
[Canonical UbuntuServer 18.04-LTS latest] Initial Kernel Version: 5.0.0-1023-azure
[Canonical UbuntuServer 18.04-LTS latest] Final Kernel Version  : 5.0.0-1023-azure
[Canonical UbuntuServer 18.04-LTS latest] Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
[Canonical UbuntuServer 18.04-LTS latest] Total Time (dd:hh:mm) : 0:0:26
[Canonical UbuntuServer 18.04-LTS latest] 
[Canonical UbuntuServer 18.04-LTS latest]    ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
[Canonical UbuntuServer 18.04-LTS latest] -------------------------------------------------------------------------------------------------------------------------------------------
[Canonical UbuntuServer 18.04-LTS latest]     1 GPU                  NVIDIA-GRID-DRIVER-VALIDATION                                                     PASS                23.86 
[Canonical UbuntuServer 18.04-LTS latest] 	Using nVidia driver : GRID 
[Canonical UbuntuServer 18.04-LTS latest] 	lsvmbus: Expected "PCI Express pass-through" count: 1, count inside the VM: 1 : PASS 
[Canonical UbuntuServer 18.04-LTS latest] 	lspci: Expected "3D controller: NVIDIA Corporation" count: 1, found inside the VM: 1 : PASS 
[Canonical UbuntuServer 18.04-LTS latest] 	lshw: Expected Display adapters: 1, total adapters found in VM: 1 : PASS 
[Canonical UbuntuServer 18.04-LTS latest] 	nvidia-smi: Expected GPU count: 1, found inside the VM: 1 : PASS 
[Canonical UbuntuServer 18.04-LTS latest] 
[Canonical UbuntuServer 18.04-LTS latest] 
[Canonical UbuntuServer 18.04-LTS latest] Logs can be found at C:\LISAv2\NH54\TestResults\2019-12-11-23-09-58-9042
```

### CODE REVIEW

File: Testscripts/Linux/gpu-driver-install.sh [hunk 1]
```diff
@@ -29,45 +28,55 @@ grid_driver="https://go.microsoft.com/fwlink/?linkid=874272"
...
 function InstallCUDADrivers() {
+    LogMsg "Starting CUDA driver installation"
     case $DISTRO in
     redhat_7|centos_7)
-        CUDA_REPO_PKG="cuda-repo-rhel7-${CUDADriverVersion}.x86_64.rpm"
-        LogMsg "Using ${CUDA_REPO_PKG}"
+        CUDA_REPO_PKG="cuda-repo-rhel7-$CUDADriverVersion.x86_64.rpm"
+        LogMsg "Using $CUDA_REPO_PKG"
 
         wget http://developer.download.nvidia.com/compute/cuda/repos/rhel7/x86_64/"${CUDA_REPO_PKG}" -O /tmp/"${CUDA_REPO_PKG}"
```
Comment (friver-ms) [inline Testscripts/Linux/gpu-driver-install.sh]: Consistency: curly brackets should be removed here too. There are several places in this file were the curly brackets are still there and should be removed

File: Testscripts/Linux/gpu-driver-install.sh [hunk 2]
```diff
@@ -29,45 +28,55 @@ grid_driver="https://go.microsoft.com/fwlink/?linkid=874272"
...
+            LogErr "Failed to download $CUDA_REPO_PKG"
             SetTestStateAborted
             return 1
+        else
+            LogMsg "Successfully downloaded the $CUDA_REPO_PKG file in /tmp directory"
         fi
 
-        rpm -ivh /tmp/"${CUDA_REPO_PKG}"
+        rpm -ivh /tmp/"$CUDA_REPO_PKG"
+        LogMsg "Installed the rpm package, $CUDA_REPO_PKG"
```
Comment (friver-ms) [inline Testscripts/Linux/gpu-driver-install.sh:47]: Should say "Installing the rpm package..." or "Starting installation of the rpm package".

File: Testscripts/Linux/gpu-driver-install.sh [hunk 3]
```diff
@@ -29,45 +28,55 @@ grid_driver="https://go.microsoft.com/fwlink/?linkid=874272"
...
             SetTestStateAborted
             return 1
+        else
+            LogMsg "Successfully downloaded $CUDA_REPO_PKG"
         fi
 
         apt-key adv --fetch-keys http://developer.download.nvidia.com/compute/cuda/repos/ubuntu"${os_RELEASE//./}"/x86_64/7fa2af80.pub
-        dpkg -i /tmp/"${CUDA_REPO_PKG}"
+        dpkg -i /tmp/"$CUDA_REPO_PKG"
+        LogMsg "Installed $CUDA_REPO_PKG"
```
Comment (friver-ms) [inline Testscripts/Linux/gpu-driver-install.sh:79]: Should say "Installing ..." or "Starting installation..."

File: Testscripts/Linux/gpu-tensorflow.sh [hunk 4]
```diff
@@ -77,59 +83,74 @@ Prepare_Test_Dependencies()
...
 
     export PATH=$(ls -d /usr/local/cuda-*)/bin${PATH:+:${PATH}}
     export LD_LIBRARY_PATH=/usr/local/cuda/lib64${LD_LIBRARY_PATH:+:${LD_LIBRARY_PATH}}
 
-    # Try to run the benchmarks test by default value.
+    # Try to run the benchmark test by default value.
     python benchmarks/scripts/tf_cnn_benchmarks/tf_cnn_benchmarks.py
     if [ $? -ne 0 ]; then
-        LogErr "Try to run benchmarks test failed"
+        LogErr "Try to run benchmark test failed"
```
Comment (friver-ms) [inline Testscripts/Linux/gpu-tensorflow.sh]: Log how the benchmark is being called

File: Testscripts/Linux/gpu-tensorflow.sh [hunk 5]
```diff
@@ -77,59 +83,74 @@ Prepare_Test_Dependencies()
...
+        LogMsg "Successfully ran benchmark test with the default value"
     fi
 }
 
-Get_Average_Utilization()
-{
+Get_Average_Utilization() {
     timeout=0
     TOTAL_TIMEOUT=300
+    LogMsg "Stablized the GPU utilization during $TOTAL_TIMEOUT. As soon as it reach out to 30% above, it is ready to test"
```
Comment (friver-ms) [inline Testscripts/Linux/gpu-tensorflow.sh]: Typo

File: Testscripts/Windows/GPU-DRIVER-INSTALL.ps1 [hunk 6]
```diff
@@ -29,13 +29,27 @@ function Start-Validation {
...
+        Write-Error "Could not succeed to fetch out PCIExpress result from lsvmbus"
+    }
+
     Set-Content -Value $PCIExpress -Path $LogDir\PCI-Express-passthrough.txt -Force
     # Scope to match GPUs only since there can be other pass-through devices
     $pciExpressCount = (Select-String -Path $LogDir\PCI-Express-passthrough.txt -Pattern "Device_ID.*47505500").Matches.Count
+    if ( $pciExpressCount -gt 0 ) {
+        Write-Debug "Successfully fetched more than a PCI Expess device "
+    } else {
+        Write-Error "Could not fetched the PCI Express device count"
```
Comment (friver-ms) [inline Testscripts/Windows/GPU-DRIVER-INSTALL.ps1]: Typo

File: Testscripts/Windows/GPU-DRIVER-INSTALL.ps1 [hunk 7]
```diff
@@ -29,13 +29,27 @@ function Start-Validation {
...
+    }
+
     if ($pciExpressCount -eq $expectedGPUCount) {
         $currentResult = $resultPass
+        Write-Debug "Successfully verified PCI Express device count with the expected GPU counts: $pciExpressCount"
```
Comment (friver-ms) [inline Testscripts/Windows/GPU-DRIVER-INSTALL.ps1]: Typo

File: Testscripts/Windows/GPU-TENSORFLOW.ps1 [hunk 8]
```diff
@@ -73,22 +76,26 @@ function Main {
...
                 -port $allVMData.SSHPort -username $user -password $password
+            Write-Debug "Collecting remote files like csv or log"
         }
 
         if ($testResult -match "PASS") {
             Write-LogInfo "Generating the performance data for database insertion"
             foreach ($param in $currentTestData.TestParameters.param) {
                 if ($param -match "CUDADriverVersion") {
                     $CUDADriverVersion = $param.Replace("CUDADriverVersion=","").Replace('"',"")
+                    Write-Debug "CUDA driver version: $$CUDADRiverVersion"
```
Comment (LiliDeng) [inline Testscripts/Windows/GPU-TENSORFLOW.ps1]: double $ for $$CUDADRiverVersion

**Review by LiliDeng — CHANGES_REQUESTED** [review]

File: Testscripts/Linux/gpu-driver-install.sh (hunk 2 above)
Comment (juhlee-microsoft) [reply Testscripts/Linux/gpu-driver-install.sh:47]: The Logging is after the command execution. So 'past' statement is correct.

File: Testscripts/Linux/gpu-driver-install.sh (hunk 3 above)
Comment (juhlee-microsoft) [reply Testscripts/Linux/gpu-driver-install.sh:79]: Again, the logging is after the command execution.

File: Testscripts/Linux/gpu-driver-install.sh (hunk 1 above)
Comment (juhlee-microsoft) [reply Testscripts/Linux/gpu-driver-install.sh]: fixed in 9626be3c614ab3d45d8977198984ea0a43941873

File: Testscripts/Linux/gpu-tensorflow.sh (hunk 4 above)
Comment (juhlee-microsoft) [reply Testscripts/Linux/gpu-tensorflow.sh]: fixed in 9626be3c614ab3d45d8977198984ea0a43941873

File: Testscripts/Windows/GPU-DRIVER-INSTALL.ps1 (hunk 6 above)
Comment (juhlee-microsoft) [reply Testscripts/Windows/GPU-DRIVER-INSTALL.ps1]: fixed in 9626be3c614ab3d45d8977198984ea0a43941873

File: Testscripts/Windows/GPU-DRIVER-INSTALL.ps1 (hunk 7 above)
Comment (juhlee-microsoft) [reply Testscripts/Windows/GPU-DRIVER-INSTALL.ps1]: fixed in 9626be3c614ab3d45d8977198984ea0a43941873

File: Testscripts/Windows/GPU-TENSORFLOW.ps1 (hunk 8 above)
Comment (juhlee-microsoft) [reply Testscripts/Windows/GPU-TENSORFLOW.ps1]: fixed in 9626be3c614ab3d45d8977198984ea0a43941873

File: Testscripts/Linux/gpu-tensorflow.sh (hunk 5 above)
Comment (juhlee-microsoft) [reply Testscripts/Linux/gpu-tensorflow.sh]: fixed in 9626be3c614ab3d45d8977198984ea0a43941873

**Comment by juhlee-microsoft** [comment]
Fixed the error and post the new test results.

[Canonical UbuntuServer 18.04-LTS latest] [LISAv2 Test Results Summary]
[Canonical UbuntuServer 18.04-LTS latest] Test Run On           : 11/13/2019 18:22:31
[Canonical UbuntuServer 18.04-LTS latest] ARM Image Under Test  : Canonical : UbuntuServer : 18.04-LTS : latest
[Canonical UbuntuServer 18.04-LTS latest] Initial Kernel Version: 5.0.0-1025-azure
[Canonical UbuntuServer 18.04-LTS latest] Final Kernel Version  : 5.0.0-1025-azure
[Canonical UbuntuServer 18.04-LTS latest] Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
[Canonical UbuntuServer 18.04-LTS latest] Total Time (dd:hh:mm) : 0:0:34
[Canonical UbuntuServer 18.04-LTS latest] 
[Canonical UbuntuServer 18.04-LTS latest]    ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
[Canonical UbuntuServer 18.04-LTS latest] -------------------------------------------------------------------------------------------------------------------------------------------
[Canonical UbuntuServer 18.04-LTS latest]     1 GPU                  NVIDIA-CUDA-DRIVER-VALIDATION                                                     PASS                 30.6 
[Canonical UbuntuServer 18.04-LTS latest] 	Using nVidia driver : CUDA 
[Canonical UbuntuServer 18.04-LTS latest] 	lsvmbus: Expected "PCI Express pass-through" count: 4, count inside the VM: 4 : PASS 
[Canonical UbuntuServer 18.04-LTS latest] 	lspci: Expected "3D controller: NVIDIA Corporation" count: 4, found inside the VM: 4 : PASS 
[Canonical UbuntuServer 18.04-LTS latest] 	lshw: Expected Display adapters: 4, total adapters found in VM: 4 : PASS 
[Canonical UbuntuServer 18.04-LTS latest] 	nvidia-smi: Expected GPU count: 4, found inside the VM: 4 : PASS 
[Canonical UbuntuServer 18.04-LTS latest] 
[Canonical UbuntuServer 18.04-LTS latest] 
[Canonical UbuntuServer 18.04-LTS latest] Logs can be found at C:\LISAv2\MQ70-20191113182225\TestResults\2019-13-11-18-22-27-6386

[Canonical UbuntuServer 18.04-LTS latest] [LISAv2 Test Results Summary]
[Canonical UbuntuServer 18.04-LTS latest] Test Run On           : 11/13/2019 18:22:49
[Canonical UbuntuServer 18.04-LTS latest] ARM Image Under Test  : Canonical : UbuntuServer : 18.04-LTS : latest
[Canonical UbuntuServer 18.04-LTS latest] Initial Kernel Version: 5.0.0-1025-azure
[Canonical UbuntuServer 18.04-LTS latest] Final Kernel Version  : 5.0.0-1025-azure
[Canonical UbuntuServer 18.04-LTS latest] Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
[Canonical UbuntuServer 18.04-LTS latest] Total Time (dd:hh:mm) : 0:0:22
[Canonical UbuntuServer 18.04-LTS latest] 
[Canonical UbuntuServer 18.04-LTS latest]    ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
[Canonical UbuntuServer 18.04-LTS latest] -------------------------------------------------------------------------------------------------------------------------------------------
[Canonical UbuntuServer 18.04-LTS latest]     1 GPU                  NVIDIA-GRID-DRIVER-VALIDATION                                                     PASS                20.28 
[Canonical UbuntuServer 18.04-LTS latest] 	Using nVidia driver : GRID 
[Canonical UbuntuServer 18.04-LTS latest] 	lsvmbus: Expected "PCI Express pass-through" count: 1, count inside the VM: 1 : PASS 
[Canonical UbuntuServer 18.04-LTS latest] 	lspci: Expected "3D controller: NVIDIA Corporation" count: 1, found inside the VM: 1 : PASS 
[Canonical UbuntuServer 18.04-LTS latest] 	lshw: Expected Display adapters: 1, total adapters found in VM: 1 : PASS 
[Canonical UbuntuServer 18.04-LTS latest] 	nvidia-smi: Expected GPU count: 1, found inside the VM: 1 : PASS 
[Canonical UbuntuServer 18.04-LTS latest] 
[Canonical UbuntuServer 18.04-LTS latest] 
[Canonical UbuntuServer 18.04-LTS latest] Logs can be found at C:\LISAv2\CY79-20191113182244\TestResults\2019-13-11-18-22-46-2217

**Comment by juhlee-microsoft** [comment]
@LiliDeng @friver-ms Please review and sign-off please

**Comment by juhlee-microsoft** [comment]
New test results

[Canonical UbuntuServer 16.04-LTS latest] [LISAv2 Test Results Summary]
[Canonical UbuntuServer 16.04-LTS latest] Test Run On           : 11/15/2019 19:06:30
[Canonical UbuntuServer 16.04-LTS latest] ARM Image Under Test  : Canonical : UbuntuServer : 16.04-LTS : latest
[Canonical UbuntuServer 16.04-LTS latest] Initial Kernel Version: 4.15.0-1063-azure
[Canonical UbuntuServer 16.04-LTS latest] Final Kernel Version  : 4.15.0-1063-azure
[Canonical UbuntuServer 16.04-LTS latest] Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
[Canonical UbuntuServer 16.04-LTS latest] Total Time (dd:hh:mm) : 0:0:14
[Canonical UbuntuServer 16.04-LTS latest] 
[Canonical UbuntuServer 16.04-LTS latest]    ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
[Canonical UbuntuServer 16.04-LTS latest] -------------------------------------------------------------------------------------------------------------------------------------------
[Canonical UbuntuServer 16.04-LTS latest]     1 GPU                  GPU-PCI-RESCIND                                                                   PASS                11.55 
[Canonical UbuntuServer 16.04-LTS latest] 	Using nVidia driver : CUDA 
[Canonical UbuntuServer 16.04-LTS latest] 	Successfully disabled and reenabled the PCI device. :  
[Canonical UbuntuServer 16.04-LTS latest] 	lsvmbus: Expected "PCI Express pass-through" count: 1, count inside the VM: 1 : PASS 
[Canonical UbuntuServer 16.04-LTS latest] 	lspci: Expected "3D controller: NVIDIA Corporation" count: 1, found inside the VM: 1 : PASS 
[Canonical UbuntuServer 16.04-LTS latest] 	lshw: Expected Display adapters: 1, total adapters found in VM: 1 : PASS 
[Canonical UbuntuServer 16.04-LTS latest] 	nvidia-smi: Expected GPU count: 1, found inside the VM: 1 : PASS 
[Canonical UbuntuServer 16.04-LTS latest] 
[Canonical UbuntuServer 16.04-LTS latest] 
[Canonical UbuntuServer 16.04-LTS latest] Logs can be found at C:\LISAv2\MX68\TestResults\2019-15-11-19-06-25-6564

**Review by friver-ms — APPROVED** [review]

**Review by LiliDeng — APPROVED** [review]

**Pull request merged**